torch/__init__.py

```diff
@@ -128,15 +128,18 @@ def is_storage(obj):
 
 
 def set_default_tensor_type(t):
-    global Storage
-    Tensor = _import_dotted_name(t)
-    Storage = _import_dotted_name(t.replace('Tensor', 'Storage'))
+    if isinstance(t, globals()['dtype']):
+        _C._set_default_tensor_type(t)
+    else:
+        global Storage
+        Tensor = _import_dotted_name(t)
+        Storage = _import_dotted_name(t.replace('Tensor', 'Storage'))
```

Comment:
This code path also sets the global `torch.Storage` which is missing from the dtype code path

Reply:
seems like this is actually a regression from when we moved the default initialization to C++.  I'll investigate.